src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java

```diff
@@ -75,26 +78,36 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
-    config.validate(stageConfigurer.getFailureCollector()); // validate before macros are substituted
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateFilters(collector);
+      return;
+    }
+    OAuthInfo oAuthInfo = SalesforceConnectionUtil.getOAuthInfo(config.getConnection().getAuthenticatorCredentials(),
+            collector);
+    config.validate(stageConfigurer.getFailureCollector(), oAuthInfo); // validate before macros are substituted
 
     // if connection params are macro or null
     if (config.getConnection() != null) {
       if (!config.getConnection().canAttemptToEstablishConnection()) {
         return;
       }
     }
-    config.validateSObjects(stageConfigurer.getFailureCollector());
+    config.validateSObjects(stageConfigurer.getFailureCollector(), oAuthInfo);
     stageConfigurer.setOutputSchema(null);
   }
 
   @Override
   public void prepareRun(BatchSourceContext context) throws ConnectionException {
     FailureCollector collector = context.getFailureCollector();
-    config.validate(collector);
-    config.validateSObjects(collector);
+    OAuthInfo oAuthInfo = SalesforceConnectionUtil.getOAuthInfo(config.getConnection().getAuthenticatorCredentials(),
+            collector);
+    config.validate(collector, oAuthInfo);
+    config.validateFilters(collector);
```

Comment:
Can we do the validations that do not require the oAuthinfo first ?

Reply:
done